tools/data-conversion/src/main.rs

```diff
@@ -549,3 +627,369 @@ fn binary_to_fixed_bit_slice(
+    };
+
+    IntermediateRepresentation {
+        sign,
+        mantissa,
+        exponent: 0,
+    }
+}
+
+/// Converts the intermediate representation to a binary string and writes it to a file or stdout.
+///
+/// This function takes an `IntermediateRepresentation`, converts the mantissa to a `BigInt`
+/// applying the sign, converts the resulting `BigInt` to a binary string, and writes
+/// the binary string to the specified file or stdout.
+///
+/// # Arguments
+///
+/// * `inter_rep` - The intermediate representation to be converted.
+/// * `filepath_send` - A mutable reference to an optional `File` where the binary string
+///   will be written. If `None`, the result is written to stdout.
+///
+/// # Returns
+///
+/// This function returns a `std::io::Result<()>` which is `Ok` if the operation
+/// is successful, or an `Err` if an I/O error occurs while writing to the file.
+
+fn intermediate_to_binary(
+    inter_rep: IntermediateRepresentation,
+    filepath_send: &mut Option<File>,
+) -> io::Result<()> {
+    let inter_value = if inter_rep.sign {
+        BigInt::from(inter_rep.mantissa)
+    } else {
+        -BigInt::from(inter_rep.mantissa)
+    };
+
+    let binary_str = inter_value.to_str_radix(2);
+
+    if let Some(file) = filepath_send {
+        file.write_all(binary_str.as_bytes())?;
+        file.write_all(b"\n")?;
+    } else {
+        std::io::stdout().write_all(binary_str.as_bytes())?;
+        std::io::stdout().write_all(b"\n")?;
+    }
+
+    Ok(())
+}
+
+/// Converts a string representation of a floating-point number to the intermediate representation.
+///
+/// This function takes a string slice representing a floating-point number,
+/// splits it into integer and fractional parts, constructs the mantissa,
+/// sets the exponent based on the length of the fractional part, and
+/// constructs an `IntermediateRepresentation` containing the sign, mantissa, and exponent.
+///
+/// # Arguments
+///
+/// * `float_string` - A string slice containing the floating-point number to be converted.
+///
+/// # Returns
+///
+/// This function returns an `IntermediateRepresentation` containing the sign, mantissa,
+/// and exponent of the floating-point number.
+///
+/// # Panics
+///
+/// This function will panic if the input string cannot be parsed as a number.
+fn float_to_intermediate(float_string: &str) -> IntermediateRepresentation {
+    let sign = !float_string.starts_with('-');
+    let float_trimmed = float_string.trim_start_matches('-');
+
+    let parts: Vec<&str> = float_trimmed.split('.').collect();
+    let integer_part = parts[0];
+    let fractional_part = if parts.len() > 1 { parts[1] } else { "0" };
+    // Prob not the best way to do this
+    let mantissa_string = format!("{integer_part}{fractional_part}");
+
+    IntermediateRepresentation {
+        sign,
+        mantissa: BigUint::from_str(&mantissa_string).expect("Invalid number"),
+        exponent: -(fractional_part.len() as i32),
```

Comment:
This appears to be assuming that the intermediate representation uses base-10 "scientific notation," but I think we want base-2?

But taking a step back, in the past we've discussed defining decimal strings as representing double-precision floats, to avoid the problems that arise when you try to handle decimal floats in an arbitrary-precision way. (And we would require the use of hexfloats if you need human-readable arbitrary-precision floats.) So I guess now would be the time to decide whether we want to commit to that strategy. If so, then you don't really need to worry about the float parsing at all; you can just defer to rust's built-in parsing, and then extract the sign, mantissa, and exponent from the `f64`.